src/daft-local-execution/src/sinks/streaming_sink.rs

```diff
@@ -129,12 +129,18 @@ impl StreamingSinkNode {
                 match result {
                     StreamingSinkOutput::NeedMoreInput(mp) => {
                         if let Some(mp) = mp {
-                            let _ = output_sender.send(mp).await;
+                            if output_sender.send(mp).await.is_err() {
```

Comment:
we should instead match on the exact error like:

`if let Err(ErrorType) = output_sender.send(mp).await`

Reply:
It's an error struct, so there's no pattern to match. And it always signifies channel closure. https://docs.rs/tokio/latest/tokio/sync/mpsc/error/struct.SendError.html